pkg/mattermost/mattermost_v1beta.go

```diff
@@ -220,13 +220,21 @@ func GenerateDeploymentV1Beta(mattermost *mmv1beta.Mattermost, db DatabaseConfig
 	volumeMounts := mattermost.Spec.VolumeMounts
 	podAnnotations := map[string]string{}
 
+	// Set user specified annotations
+	if mattermost.Spec.PodTemplate.Annotations != nil {
+		podAnnotations = mattermost.Spec.PodTemplate.Annotations
+	}
+
 	// Mattermost License
 	if len(mattermost.Spec.LicenseSecret) != 0 {
 		env, vMount, volume, annotations := mattermostLicenceConfig(mattermost.Spec.LicenseSecret)
 		envVarGeneral = append(envVarGeneral, env)
 		volumeMounts = append(volumeMounts, vMount)
 		volumes = append(volumes, volume)
-		podAnnotations = annotations
+		// Add prometheus annotations, overwriting user specified if needed
+		for k, v := range annotations {
+			podAnnotations[k] = v
+		}
```

Comment:
Also adding another test for Deployment generation would be much appreciated. On top of the labels and annotations application order, we could verify that the additional `podTemplate.Labels` are applied only to the pods, not all other resources.

Reply:
Added - I'm not sure if adding this check the way I did is a good approach or not. Effectively just added a check for the service (no other resources) and placed it in the other generate service tests.